### PR TITLE
Milestone 2 - Prepare project structure, display and add buttons

### PR DESCRIPTION
**Pull request by RyelBanfield**
This PR contains everything required for milestone 2 of the react calculator project.

### CODE REVIEW

**Comment by bellom** [comment]
> This PR contains everything required for milestone 2 of the react calculator project.

It would be nice to be more specific while writing your PR description, list out the basic features you added.
And also, try to commit more often.

File: src/components/App.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const result = '0';
+
+const App = () => (
+  <div className="App">
+    <header className="App-header">
+      Calculator built using React
+    </header>
+    <Display result={result} />
+    <ButtonPanel />
+  </div>
```
Comment (bellom) [inline src/components/App.js:13]: You need to wrap both `Display` and `ButtonPanel` with [React.Fragment](https://reactjs.org/docs/fragments.html#short-syntax)
Reason:
We don't want to create an extra element, that's why you need to use fragment, so we are able to render without having to create an extra node wrapping `Display` and `ButtonPanel`.

File: src/components/ButtonPanel.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      <Button buttonName="1" />
+      <Button buttonName="2" />
+      <Button buttonName="3" />
+      <Button buttonName="+" />
+    </div>
+    <div>
+      <Button buttonName="0" />
+      <Button buttonName="." />
+      <Button buttonName="=" />
+    </div>
```
Comment (bellom) [inline src/components/ButtonPanel.js:33]: ### [Optional]
Your code appears to be WET-(Write Every Time), Microverse encourage DRY-(Don't Repeat Yourself) code, In your case; you repeated the `Button` component multiple times.

You can use [objects.key](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys)  and `map` to loop through the objects. `objects.keys` for the outer loop and `map` for the inner loop.

Example; 
```
const GroupButtons = {
  Group 1: ['AC', '+/-', '%', '÷'],
  Group 2: ['7', '8', '9', 'X'],
   ............
}
```

File: src/components/Display.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import PropTypes from 'prop-types';
+
+const Display = ({ result }) => (
+  <div>
+    {result}
+  </div>
+);
+
```
Comment (bellom) [inline src/components/Display.js:8]: Oops, Let's try to add default props, which will be displayed once we open the app just like a normal calculator.

The result should have a default value.
Display.defaultProps = `{ result: '0' }`

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @RyelBanfield  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best**

**Review by oloomoses — APPROVED** [review]
## Congratulations, Project Approved. :muscle: :tada: :clap: 
**Hello @RyelBanfield,
Great job working on the changes suggested by @bellom.
You have passed all the requirements for this milestone :muscle:**

Feel free to merge and close this pull request.

**I wish you the best in your next milestone and see you at the top :rocket:** 
#### Happy Coding
_Slack: Oloo Moses_

**Pull request merged**